csrc/kernel.cpp

```diff
@@ -425,6 +425,9 @@ void Kernel::finalize(std::vector<Expr*> top_level_exprs) {
   parameters_ = GpuLower::current()->allKnownVals();
   parameters_.insert(parameters_.end(), outputs().begin(), outputs().end());
   for (auto alloc : summary_.global_allocations) {
+    if (alloc->buffer()->isFusionOutput() || alloc->alias() != nullptr) {
```

Comment:
Since even fusion outputs now have `kir::Allocate`, they show up in `globalallocations`, which need to be ignored here as they are already added at line 426.

Alternatively, we could remove line 426 and add all of outputs and intermediates together here, but that would change the ordering of kernel parameters. That should be fine, but just to minimize any unnecessary changes.